### ISSUE TITLE
Referencing smart constructor from the companion object in trait definition doesn't compile

### ISSUE THREAD

**Issue by FabriceCastel** [fpinscala/fpinscala#508]
So I was unable to get the answer for ex 5.3. I checked the answer key and pasted that into my trait def and it wouldn't compile! The compiler complained that it couldn't resolve the symbols `empty` and `cons`. So I tried instead using `Stream.cons` and `Stream.empty` and that worked.

Why did I have to do this? Why isn't this in the answer key?

**Comment by arobson73** [comment]
do you have import Stream._ at top of file ?
how did the take function above it compile since that also uses empty and cons.
i take it your using sbt, try reloading it (reload)  ?

**Comment by FabriceCastel** [comment]
Yeah, at the top of my file I did import Stream, but to prevent it from being a default import (which I believe it is?). I thought this would be relatively harmless.

`import scala.collections.immutable.{Stream => _}`

There might be a typo there, I don't have the code in front of me..

Not sure which function you mean when you say the one "above it," it's very possible that I skipped implementing another function that required some of the things defined in the companion object.

I'm using IntelliJ with a worksheet.

**Comment by arobson73** [comment]
have a look in the answers folder for that code (on github), the take function for exercise 5.2 is the function I meant. If that compiles, then you should have no problem with takeWhile. 

they just use 
import Stream._

this means that instead of having to prefix Stream.empty or Stream.cons you can drop the Stream._ and just use import Stream._ instead.

the stream that you have prevented is the collections Stream. (not the same thing)

i tend to use sbt now (for compiling and running), and use intelliJ for browsing code only.
